oak-store-document/src/test/java/org/apache/jackrabbit/oak/plugins/document/mongo/MongoConnectionTest.java

```diff
@@ -139,4 +147,156 @@ private MongoClient mockMongoClient(boolean replicaSet) {
+        int maxWaitTime = 45000;
+        int maxConnectionLifeTime = 61442;
+        int maintenanceInitialDelay = 2234;
+        int maintenanceFrequency = 28102;
+        int heartbeatFrequency = 7890;
+        int maxIdleTime = 18543;
+        int minPoolSize = 12;
+        int maxPoolSize = 67;
+
+        String testUri = "mongodb://localhost:27017/testdb";
+        ConnectionString mongoURI = new ConnectionString(testUri);
+
+        MongoClientSettings mongoSettings = MongoClientSettings.builder()
+                .applyConnectionString(mongoURI)
+                .applyToSocketSettings(builder -> builder
+                        .connectTimeout(connectTimeout, TimeUnit.MILLISECONDS)
+                        .readTimeout(socketTimeout, TimeUnit.MILLISECONDS))
+                .applyToConnectionPoolSettings(builder -> builder
+                        .maxWaitTime(maxWaitTime, TimeUnit.MILLISECONDS)
+                        .maxConnectionLifeTime(maxConnectionLifeTime, TimeUnit.MILLISECONDS)
+                        .maintenanceInitialDelay(maintenanceInitialDelay, TimeUnit.MILLISECONDS)
+                        .maintenanceFrequency(maintenanceFrequency, TimeUnit.MILLISECONDS)
+                        .maxConnectionIdleTime(maxIdleTime, TimeUnit.MILLISECONDS)
+                        .minSize(minPoolSize)
+                        .maxSize(maxPoolSize))
+                .applyToServerSettings(builder -> builder
+                        .heartbeatFrequency(heartbeatFrequency, TimeUnit.MILLISECONDS))
+                .build();
+
+        ConnectionPoolSettings mongoPoolSettings = mongoSettings.getConnectionPoolSettings();
+        ServerSettings mongoServerSettings = mongoSettings.getServerSettings();
+
+        // Assert programmatically set parameters
+        assertEquals(connectTimeout, mongoSettings.getSocketSettings().getConnectTimeout(TimeUnit.MILLISECONDS));
+        assertEquals(socketTimeout, mongoSettings.getSocketSettings().getReadTimeout(TimeUnit.MILLISECONDS));
+        assertEquals(maxWaitTime, mongoPoolSettings.getMaxWaitTime(TimeUnit.MILLISECONDS));
+        assertEquals(maxConnectionLifeTime, mongoPoolSettings.getMaxConnectionLifeTime(TimeUnit.MILLISECONDS));
+        assertEquals(maintenanceInitialDelay, mongoPoolSettings.getMaintenanceInitialDelay(TimeUnit.MILLISECONDS));
+        assertEquals(maintenanceFrequency, mongoPoolSettings.getMaintenanceFrequency(TimeUnit.MILLISECONDS));
+
+        // Assert the 4 parameters that were also in the connection string test
+        assertEquals(heartbeatFrequency, (int) mongoServerSettings.getHeartbeatFrequency(TimeUnit.MILLISECONDS));
+        assertEquals(maxIdleTime, (int) mongoPoolSettings.getMaxConnectionIdleTime(TimeUnit.MILLISECONDS));
+        assertEquals(minPoolSize, mongoPoolSettings.getMinSize());
+        assertEquals(maxPoolSize, mongoPoolSettings.getMaxSize());
+
+        if (DEBUG) {
+            System.out.println("=== testMongoDBDriverProgrammaticParameters ===");
```

Comment:
why don't we use a logger here.
Also, the assertEquals above, would print if the values are not equal.